rewrite-java/src/main/java/org/openrewrite/java/ReplaceStringLiteralWithConstant.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+
+
+import java.util.List;
+
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class ReplaceStringLiteralWithConstant extends Recipe {
+
+    @Option(displayName = "Owning type of the constant",
+            example = "org.springframework.http.MediaType")
+    String owningType;
+
+    @Option(displayName = "Constant name",
+            example = "APPLICATION_JSON_VALUE")
+    String constantName;
+
+    @Option(displayName = "Literal value",
+            example = "application/json")
+    String literalValue;
```

Comment:
We might want to make this value optional, and default the to value of the constant that should replace the literal. That ensures we can configure this recipe with a minimal number of configuration options.